### PR TITLE
ACAS-699: Accept projects list to filter /api/experiments/protocolCodename/:code on from from nodejs

### PR DESCRIPTION
**Pull request by brianbolt**
## Description
 - Implement filtering on projects for route /api/experiments/protocolCode name

## Related Issue
ACAS-699

## How Has This Been Tested?
acasclient tests

### CODE REVIEW

File: src/main/java/com/labsynch/labseer/api/ApiExperimentController.java [hunk 1]
```diff
@@ -1634,20 +1635,19 @@ public ResponseEntity<java.lang.String> jsonFindExperimentByNameGet(@RequestPara
 	@RequestMapping(value = "/protocol/{codeName}", method = RequestMethod.GET, headers = "Accept=application/json")
 	@ResponseBody
 	public ResponseEntity<java.lang.String> jsonFindExperimentsByProtocolCodeName(
-			@PathVariable("codeName") String codeName, @RequestParam(value = "with", required = false) String with) {
+			@PathVariable("codeName") String codeName, @RequestParam(value = "projects", required = false) List<String> projects, @RequestParam(value = "with", required = false) String with) {
```
Comment (brianbolt) [inline src/main/java/com/labsynch/labseer/api/ApiExperimentController.java:1638]: New projects query param and removed most of the logic from the api controller an moved it to a new service.

File: src/main/java/com/labsynch/labseer/service/ExperimentServiceImpl.java [hunk 2]
```diff
@@ -1118,6 +1118,45 @@ public Collection<Experiment> findExperimentsByGenericMetaDataSearch(String quer
 
 	}
 
+	public Collection<Experiment> findExperimentsByProtocolCodeName(String codeName, List<String> projects)
```
Comment (brianbolt) [inline src/main/java/com/labsynch/labseer/service/ExperimentServiceImpl.java:1121]: New service which will handle filter the experiments like some of our other services already do it.

**Review by hhan-schrodinger — APPROVED** [review]
Looks good to me!

File: src/main/java/com/labsynch/labseer/service/ExperimentServiceImpl.java [hunk 3]
```diff
@@ -1118,6 +1118,45 @@ public Collection<Experiment> findExperimentsByGenericMetaDataSearch(String quer
...
+			throws TooManyResultsException, NoResultException {
```
Comment (hhan-schrodinger) [inline src/main/java/com/labsynch/labseer/service/ExperimentServiceImpl.java:1122]: Just curious - how would TooManyResultsException get thrown here?

Comment (brianbolt) [reply src/main/java/com/labsynch/labseer/service/ExperimentServiceImpl.java:1122]: Good point! It doesn't, I accidentally throw a RunTimException instead.

Comment (brianbolt) [reply src/main/java/com/labsynch/labseer/service/ExperimentServiceImpl.java:1122]: Fixed

**Pull request merged**